packages/server/src/restaurants/infra/DishesFacade.ts

```diff
@@ -0,0 +1,14 @@
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+
+import { Dish, DishDocument } from '../dishes/database';
+
+class DishesFacade {
+  constructor(@InjectModel(Dish.name) private readonly dishModel: Model<DishDocument>) {}
+
+  async findById(id: string) {
+    return this.dishModel.findById(id);
+  }
```

Comment:
W tym miejscu gdzie jest używana ta funkcja, chyba nie potrzebujemy całego dish? Wystarczy tylko sprawdzić czy dish o podanym dishId istnieje.

Reply:
Jest raczej potrzebne, bo [tutaj](https://github.com/CodersCamp2021-HK/CodersCamp2021.Project.Fullstack/pull/105/files#diff-cac72d120f18a8d9a78da62ddbd0c3f28128364971e9a03f14819820266f300cR29) jest potem zwracane dodane danie. Alternatywnie mogę zwracać 204 i wtedy to faktycznie może być boolem.

Reply:
User jak dodaje do ulubionych, to zapewne ma otwarty przed sobą widok dish, a ponieważ operacja `PUT users/favourite/dishes` (swoją drogą może lepiej `POST`) nie mutuje dish, to nie ma sensu zwracać mu dish (bo nie uległ zmianie) - zaoszczędzimy tym samym trochę mocy obliczeniowej (serializacja) i sieci (dish payload). Więc lepiej wysłać 204.

Reply:
Ma sens, ograniczyłem do `exists` i zmieniłem na `204`.

Co do metody, sugerowałem się tym, że ([w teorii](https://datatracker.ietf.org/doc/html/rfc7231#section-8.1.3)) PUT jest idempotentny, a POST nie, więc (w teorii) PUT powinien gwarantować nam, że robiąc dwa razy request z tym samym id, na końcowej liście będzie tylko jeden entry dla tego id, a nie dwa, gdzie POST mógłby dodać duplikatowy wpis. No a w praktyce to możemy nazwać wszystko jak chcemy i to kwestia konwencji jakie przyjmiemy 👍

Reply:
Masz rację z PUT-em :+1: